design/cdk-bootstrap.md

```diff
@@ -0,0 +1,199 @@
+  (CreateChangeSet and ExecuteChangeSet) in the continuous delivery pipeline.
+  It is also assumable by any principal from the account(s) passed by the `trust` option.
+  
+  This role has the following permissions:
+  
+  ```json
+  {
+    "PolicyDocument": {
+      "Statement": [
+        {
+          "Action": [
+            "cloudformation:CreateChangeSet",   "cloudformation:DeleteChangeSet",
+            "cloudformation:DescribeChangeSet", "cloudformation:DescribeStacks",
+            "cloudformation:ExecuteChangeSet"
+          ],
+          "Resource": "*"
+        },
+        {
+          "Action": "iam:PassRole",
+          "Resource": "arn-of-the-cfn-deployment-role"
+        }
+      ]
+    }
+  }
+  ```
+
+* An IAM role, called the **CFN deployment role**,
+  that will be used to perform the actual CFN stack deployment in the continuous delivery pipeline to this environment.
+  It is assumable *only* by the CloudFormation service principal
+  (this is for security reasons, as this role will have, necessarily, very wide permissions).
+  It will not have any inline policies,
+  but will instead have the Managed Policies attached that the user selected when prompted (see above).
+
+#### Physical resource names
+
+All of the above resources will be created with well-defined physical names -
+none of them will rely on automated CloudFormation naming.
+This is so that the other stages of CDK synthesis,
+like asset resolution, can rely on the concrete names
+(there is no reliable way to transfer this kind of information in an automated way across regions and/or accounts).
+This also allows for more fine-grained permissions -
+for instance, the continuous delivery pipeline needs to grant the **action role**
+permissions to read from the pipeline's bucket
+(to read the artifact that contains the CFN template to deploy);
+this way, it can add a well-defined name to the resource policy of the artifact bucket,
+instead of granting those permissions to all principals in the account.
+
+The naming scheme will include the following elements in order to minimize the chance of name collisions:
+
+* The region we're bootstrapping in.
+* The account ID we're bootstrapping in.
+* The type of the resource (file assets bucket, Docker assets repository, etc.).
+* An optional *qualifier* that defends against S3 bucket sniping.
+
+### CLI options in detail
+
+These options are inherited from the current CLI experience:
+
+* `--profile`: use the given local AWS credentials profile when interacting with the target environment.
+
+* `--stack-name`, `-s`: allows you to explicitly name the CloudFormation bootstrap stack
+  (instead of relying on the default naming scheme).
+
+* `--bucket-name`, `-b`: allows you to explicitly name the file assets S3 bucket
+  (instead of relying on the default naming scheme).
+
+* `--kms-key-id`, `-k`: optional identifier of the KMS key used for encrypting the file assets S3 bucket.
+
+These are the new options:
```

Comment:
No need to separate old and new

Reply:
I kind of like it. Makes it easy at a glance to see what is there because of backwards compatibility, and what we're adding.

Reply:
You can add a section in the end that describes the delta from the current experience.